src/Altinn.Notifications.Core/Services/OrderRequestService.cs

```diff
@@ -53,11 +55,16 @@ public async Task<(NotificationOrder? Order, ServiceError? Error)> RegisterEmail
         return (savedOrder, null);
     }
 
-    private List<INotificationTemplate> SetFromAddressIfNotDefined(List<INotificationTemplate> templates)
+    private List<INotificationTemplate> SetSenderIfNotDefined(List<INotificationTemplate> templates)
     {
         foreach (var template in templates.OfType<EmailTemplate>().Where(template => string.IsNullOrEmpty(template.FromAddress)))
         {
-            template.FromAddress = _defaultFromAddress;
+            template.FromAddress = _defaultEmailFromAddress;
+        }
+
+        foreach (var template in templates.OfType<SmsTemplate>().Where(template => string.IsNullOrEmpty(template.SenderHandle)))
+        {
+            template.SenderHandle = _defaultSmsSender;
         }
 
```

Comment:
I think this could be refactored so the list will be looped only once and still keep the readability.

Reply:
1. Firstly, I'm a bit curious how that can be achieved 🤔 Are you thinking about using conditionals? An example would be really helpful. 
2. Secondly, how running a single loop is better in this context? 🤔 So far, in this context, we are querying the templates to get only those which has empty property, and then these goes in the loop. Therefore, isn't running one loop will still go through the same number of elements as running two loops? 🤔

Reply:
I was a bit fast when I wrote this comment. Linq will of course filter out the temlates. 
So it all will come to readability, not performance.
I thought about using conditions but I tried to refactor and the result was readable but more code. So keep it as it is.

Example jusf for "fun":
foreach (var template in templates)
{
    switch (template)
    {
        case EmailTemplate emailTemplate when string.IsNullOrEmpty(emailTemplate.FromAddress):
            emailTemplate.FromAddress = _defaultEmailFromAddress;
            break;
        case SmsTemplate smsTemplate when string.IsNullOrEmpty(smsTemplate.SenderHandle):
            smsTemplate.SenderHandle = _defaultSmsSender;
            break;
    }
}

Reply:
In this case both loops goes through all elements in the list at least once. LINQ doesn't remember which elements that were taken care of in the previous iteration and would need to start over. 

Logically this code would generate four loops, but because of deferred execution and iterator support in LINQ and in foreach there is probably just two loops.